packages/editor/src/components/writing-flow/index.js

```diff
@@ -49,6 +49,29 @@ const isTabbableTextField = overEvery( [
 	focus.tabbable.isTabbableIndex,
 ] );
 
+/**
+ * Returns true if the element should consider edge navigation upon a keyboard
+ * even of the given directional key code, or false otherwise.
+ *
+ * @param {Element} element     HTML element to test.
+ * @param {number}  keyCode     KeyboardEvent keyCode to test.
+ * @param {boolean} hasModifier Whether a modifier is pressed.
+ *
+ * @return {boolean} Whether element should consider edge navigation.
+ */
+export function isNavigationCandidate( element, keyCode, hasModifier ) {
+	const isVertical = ( keyCode === UP || keyCode === DOWN );
+
+	// Currently, all elements support unmodified vertical navigation.
+	if ( isVertical && ! hasModifier ) {
+		return true;
+	}
+
+	// Native inputs should not navigate horizontally.
+	const { tagName } = element;
+	return tagName !== 'INPUT' && tagName !== 'TEXTAREA';
```

Comment:
Would it make sense to apply this to all the editables?

Reply:
Can you elaborate on what you mean? Are you thinking to reverse this condition so that we test whether the candidate is explicitly contentEditable ? Or do you mean we should include editables In the excluded set of candidates? That would cause issues for navigating between paragraphs, where we do want ArrowRight to navigate between blocks.